src/MPGDDIRC_geo.cpp

```diff
@@ -0,0 +1,243 @@
+  xml_dim_t                               mpgd_dirc_pos = x_det.position();
+  Assembly 				  assembly(det_name);
+
+  map<string, std::array<double, 2>> module_thicknesses;
+  sens.setType("tracker");
+
+  // loop over the modules
+  for (xml_coll_t mi(x_det, _U(module)); mi; ++mi) {
+    xml_comp_t x_mod = mi;
+    string     m_nam = x_mod.nameStr();
+
+    if (volumes.find(m_nam) != volumes.end()) {
+      printout(ERROR, "MPGDDIRC_geo",
+               string((string("Module with named ") + m_nam + string(" already exists."))).c_str());
+      throw runtime_error("Logics error in building modules.");
+    }
+
+    int    ncomponents     = 0;
+    double total_thickness = 0;
+
+    // Compute module total thickness from components
+    xml_coll_t ci(x_mod, _U(module_component));
+    for (ci.reset(), total_thickness = 0.0; ci; ++ci) {
+      total_thickness += xml_comp_t(ci).thickness();
+    }
+    // the module assembly volume
+    Assembly m_vol(m_nam);
+    volumes[m_nam] = m_vol;
+    m_vol.setVisAttributes(description, x_mod.visStr());
+
+    // Optional module frame.
+    // frame is 4  bars around the perimeter of the rectangular module. The frame will eat the
+    // overlapping module area
+    //
+    //  ___
+    // |___| <-- example module cross section (x-y plane), frame is flush with the
+    //           bottom of the module and protrudes on the top if needed
+    
+    // Get frame width, as it impacts the main module for being built. We
+    // construct the actual frame structure later (once we know the module width)
+    double frame_width = 0;
+    if (x_mod.hasChild(_U(frame))) {
+      xml_comp_t m_frame = x_mod.child(_U(frame));
+      frame_width        = m_frame.width();
+    }
+
+    double thickness_so_far    = 0.0;
+    double thickness_sum       = -total_thickness / 2.0;
+    double max_component_width = 0;
+    double max_component_length = 0;
+    double gas_thickness = 0.0;
+    for (xml_coll_t mci(x_mod, _U(module_component)); mci; ++mci, ++ncomponents) {
+      xml_comp_t x_comp = mci;
+      string     c_nam  = _toString(ncomponents, "component%d");
+      string  comp_name = x_comp.nameStr();
+
+      double box_width = x_comp.width();
+      double box_length = x_comp.length();
+      Box c_box;
+      //Since MPGD frames are layed over the MPGD foils, the foil material is pressent under the frame as well. 
+      //The gas volumes are not present under the frames, so our frames must eat only the gas module areas
+      //
+      //  ------------------- MPGD foil 
+      //  --               -- Frame
+      //  --  gas volume   -- Frame
+      //  --               -- Frame
+      //  ------------------- MPGD foil
+
+      //Look for gas modules to subtract frame thickness from
+      //FIXME: these module names are hard coded for now. Should find 
+      //a way to set a arribut via the moduel tag to flag what components 
+      //need to have frame thickness subtracted.
```

Comment:
Is this why this is implemented as a different geometry plugin? Mainly I'm hoping we can avoid code duplication between this and the tracker geometry plugin.

Reply:
This is one reason. Another is that this is the only tracking code which uses rectangular geometry. The other right now I believe use  cylindrical geometries.